### PR TITLE
Refactoring: removing the global for signal mask in sched

### PR DESCRIPTION
**Pull request by agrawalravi90**
#### Describe Bug or Feature

Recently I fixed a signal blocking issue with scheduler multi-threading (https://github.com/openpbs/openpbs/pull/2376), but I added a global dependency on libsched in the process, which can cause issues to anything that links with libsched other than pbs_sched, and is a bad coding practice as well. So, this PR refactors the original fix.

### CODE REVIEW

**Review by arungrover — APPROVED** [review]
LGTM

File: src/scheduler/multi_threading.cpp [hunk 1]
```diff
@@ -240,7 +238,7 @@ worker(void *tid)
 	ntid = *(int *)tid;
 
 	/* Add HUP to the list of signals to block, if we ever unblock this, we'll need to modify 'restart()' to handle MT */
-	set = allsigs;
+	pthread_sigmask(SIG_BLOCK, NULL, &set);
```
Comment (arungrover) [inline src/scheduler/multi_threading.cpp:241]: Just a minor comment, shouldn't this be SIG_SETMASK? something like pthread_sigmask(SIG_SETMASK, &set, NULL)?

Comment (agrawalravi90) [reply src/scheduler/multi_threading.cpp:241]: As we discussed Arun, when 'set' is NULL, the first argument is ignored, so it doesn't really matter.

Comment (arungrover) [reply src/scheduler/multi_threading.cpp:241]: Nevermind, I was confused by how man page has explained SIG_SETMASK.  https://www.gnu.org/software/libc/manual/html_node/Process-Signal-Mask.html explains better. Please ignore the comment.

Comment (subhasisb) [reply src/scheduler/multi_threading.cpp:241]: I think u could do this in one pthread_sigmask() call right? SIG_BLOCK is supposed to "add" to the current mask-set
By doing this:
sigemptyset(&set);
sigaddset(&set, SIGHUP);
pthread_sigmask(SIG_BLOCK, &set, NULL);

**Review by subhasisb — CHANGES_REQUESTED** [review]

**Review by subhasisb — APPROVED** [review]

**Review by arungrover — APPROVED** [review]

**Pull request merged**